### DISCUSSION TITLE
Simplify installation device selection

### DISCUSSION THREAD

**Discussion by joseivanlopez** [agama-project/agama#776] [Ideas]
The Agama UI can be a bit confusing when using LVM. Let's say you have selected */dev/vdb* as the device to install on (a.k.a. installation device).  

![Screenshot from 2023-09-27 15-50-10](https://github.com/openSUSE/agama/assets/1112304/65e00a3b-6d9c-4698-9411-5298f491d4e0)

And then you activate LVM and click on the LVM settings icon in order to configure the System Volume Group. In the popup, you select *Custom devices* and */dev/vdc* as the target device for creating the volume group. 

![Screenshot from 2023-09-27 15-50-57](https://github.com/openSUSE/agama/assets/1112304/1043e239-fc1d-479e-af64-a4f1311c804d)

As the popup states, every file system is going to be created as a logical volume of the volume group, that is, */dev/vdc* would be used for the installation. So what the installation device */dev/vdb* really means in this case?

Answer: the installation device is only used as boot device, that is, the device in which everything related to booting is created (boot partition if needed, install bootloader, etc).

This behaviour could be surprising (even unexpected) and difficult to explain in the UI.

### How to solve the problem

We could get rid of all this confusion by unifying the installation device selection and the LVM config concepts in a single selection. The idea is to remove the separate widget for configuring LVM, having a single setting for selecting the device to install on.

![Screenshot from 2023-09-27 16-26-53](https://github.com/openSUSE/agama/assets/1112304/7e15a3bb-b54e-4de0-9eb0-73c32ea590e7)

As before, you can still change the target disk by clicking on the device name. But now the popup would allow you to select a single disk or install using LVM. 

![installation-device-selector](https://github.com/openSUSE/agama/assets/1112304/c7abf2b1-d4b8-4cdd-a6cb-976511536fe2)

This new selection makes very clear what the target device(s) is: either the selected disk or the set of disks selected for creating the system volume group.

If LVM was selected, then the installation device would contain "LVM System Volume Group" instead of  the info of a single device.

![Screenshot from 2023-09-27 16-24-00](https://github.com/openSUSE/agama/assets/1112304/e0bdd07d-de8f-48c9-9011-4ae2796b5b18)

**Comment by shundhammer** [comment]
The term "installation device" confused me for a while - I thought at first it was the installation medium. But in reality, you mean "device to install on" - that's different. It might be a source of confusion for our users, too.

And we also tend to over-generalize those terms... wouldn't that be the "disk to install on" for most users? IMHO it's important to give users something to grip on; some well-known terms. Even if that comes at the cost of not being 100% exact in some very exotic cases, like when it's maybe not a disk but a SciFi HoloCrystalStorageCube or whatever.

**Comment by shundhammer** [comment]
I like it in general. Just some small remarks:

- A user with no previous Linux experience might need some explanation what the various partitions are. Maybe we could add something like

  - /    System
  - /home  User's personal data and work space
  - swap  Virtual memory on disk to extend the RAM
  - /var/log/audit  Protocol data for the audit subsystem

  If there is not enough space on the screen to display it directly, tooltips come to mind.


- I like the disk selection with the disk icon and the description what's on the disk. But when a disk is empty, we could tell the user. And displaying the partition table under "Content" is a bit misleading; users might think "OMG, there is still some old MS-DOS on that disk, wasting space. How do I get rid of that stuff?"

  Maybe we should rename that here to something like "Legacy PC partition table (MS-DOS style)" or "Legacy PC (MS-DOS) partition table". Just a thought.

- Do we show the user when a disk is used (largely or completely) as an LVM physical volume? I would think and hope so.

**Comment by joseivanlopez** [comment]
BTW, the summary page is not adapted yet. It only shows the device selected as installation device (boot device), even though the system volume group is using other devices.

![Screenshot from 2023-09-27 16-55-15](https://github.com/openSUSE/agama/assets/1112304/9018c899-10c9-45ab-8456-fd378b736efc)

**Reply by dgdavid** [reply]
Yep, that's something I noticed yesterday while testing the ahama-live ISO and I put it in my notes for the daily call. Good that you bring it here instead 👍 

In general, the summary should be improved. I'd like to see there if encryption is in use too.

**Comment by joseivanlopez** [comment]
In short:

* The UI is confusing.
* The installation summary does not inform about the devices used for the system volume group.
* The content of the device selected as installation device (boot device) is totally removed, even though that device is not actually used for creating any partition.

**Comment by ancorgs** [comment]
This aspect is covered by https://github.com/openSUSE/agama/pull/1031

**Comment by dgdavid** [comment]
In latest months the Storage page UI has changed a lot and the main problem described here is no longer present. Let's close it as resolved.
